[cherry-pick][debugserver] Mark ASAN memory regions as "heap"

felipepiovezan opened the pull request:
This memory type is currently not handled, but it makes sense to mark it as a heap allocation in requests asking for memory region info.

jasonmolenda reviewed (approved):
LGTM but this should be landed upstream shouldn't it?

felipepiovezan commented:
> LGTM but this should be landed upstream shouldn't it?

Oh yeah, good point.

felipepiovezan committed:
[debugserver] Mark ASAN memory regions as "heap" (#113968)

This memory type is currently not handled, but it makes sense to mark it
as a heap allocation in requests asking for memory region info.

(cherry picked from commit 474234a09655e57b7a4270150f0926db77e864b4)

felipepiovezan commented:
Transformed this into a cherry-pick of the patch landed upstream.

felipepiovezan commented:
@swift-ci test
